app/controllers/products_controller.rb

```diff
@@ -8,4 +8,16 @@ def show
                              .last
                              .price_in_brl
   end
+  
+  def update_status
+    return unless admin_signed_in?
+    product = Product.find(params[:id])
+    if product.update(status: params[:status])
+      redirect_to product, notice: 'Status atualizado com sucesso'
+    else
+      error_details = product.errors.full_messages.join(', ').downcase
+      redirect_to product, alert: "Houve um erro. Para colocar um produto à venda, #{error_details}."
+    end
```

Comment:
```suggestion
if product.update(status: params[:status])
  return redirect_to product, notice: 'Status atualizado com sucesso'
end
error_details = product.errors.full_messages.join(', ').downcase
redirect_to product, alert: "Houve um erro. Para colocar um produto à venda, #{error_details}."
```

Fiz uma sugestão, mas também é possível usar Guard clause. https://campuscode.com.br/conteudos/guard-clause-em-ruby

Reply:
então, eu sei que a comunidade não gosta muito de if/else, mas eu acho que fica mais claro com ele.

eu usaria guard clause quando a coisa testada quebra tudo (tipo não vem o product_id). mas nesse caso que é um comportamento numa condição, outro na condição contrária, acho mais coerente com if/else